### DISCUSSION TITLE
How do I use Roslyn to generate LSIF indexes?

### DISCUSSION THREAD

**Discussion by garobrik** [dotnet/roslyn#50290] [Q&A]
Hullo, I'm trying to generate LSIF indexes with Roslyn and was hoping for some help with an error I'm seeing. I'm using the tool found [here](https://github.com/dotnet/roslyn/tree/master/src/Features/Lsif/Generator) using the following invocation from the project root:
```sh
dotnet run -p src/Features/Lsif/Generator/Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.csproj --solution Roslyn.sln
```
I'm getting the following error:
```sh
System.ComponentModel.Win32Exception (2): No such file or directory
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at Microsoft.DotNet.Cli.Utils.Command.Execute(Action`1 processStarted)
   at Microsoft.DotNet.Cli.Utils.Command.Execute()
   at Microsoft.DotNet.Tools.Run.RunCommand.Execute()
   at Microsoft.DotNet.Tools.Run.RunCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```
I'm running Ubuntu 20.04, and I see the same error whether I use `dotnet 5.0.101` or `dotnet 5.0.100`.

Any help would be appreciated!

**Comment by jasonmalinowski** [comment]
At the moment the project is marked as targeting net472 which would mean it's targeting the Windows-only .NET Framework. I'm wondering if that means something isn't deploying correctly. If you change the line [here](https://github.com/dotnet/roslyn/blob/1bb65b0645ede1f8cac0d3a5b7f1bdea23cba705/src/Features/Lsif/Generator/Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.csproj#L9) to `net5.0` that should switch the build over; I don't _think_ there would be problematic dependencies but I can't say off the top of my head.

**Reply by garobrik** [reply]
Thanks for getting back to me!

After the switch, I'm seeing this error:
```
/usr/share/dotnet/sdk/5.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.DefaultItems.targets(362,5): error NETSDK1136: The target platform must be set to Windows (usually by including '-windows' in the TargetFramework property) when using Windows Forms or WPF, or referencing projects or packages that do so. [src/Features/Lsif/Generator/Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.csproj]

The build failed. Fix the build errors and run again.
```
Quick search said to fix this by substituting net5.0-windows in the same place which I gave a shot even though I'm on linux, and that yielded:
```
/usr/share/dotnet/sdk/5.0.101/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(324,5): error NETSDK1073: The FrameworkReference 'Microsoft.WindowsDesktop.App.WPF' was not recognized [/home/arrow/sourcegraph/roslyn/roslyn2/src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj]

The build failed. Fix the build errors and run again.
```

I know I'm off the beaten path here but any ideas would be appreciated!
